credentials/oauth/oauth.go

```diff
@@ -61,6 +61,21 @@ func (ts TokenSource) GetRequestMetadata(ctx context.Context) (map[string]string
 	}, nil
 }
 
+// oauthAccess supplies credentials from a given token.
+type oauthAccess struct {
+	token *oauth2.Token
+}
+
+func NewOauthAccess(token *oauth2.Token) credentials.Credentials {
```

Comment:
err, it is okay to take pointer here if it is convenient for impl. Then just change the following "token" to "*token".